completions/dpkg

```diff
@@ -1,37 +1,62 @@
 # dpkg(1) and related commands completion                  -*- shell-script -*-
 
-if _comp_have_command grep-status; then
-    _comp_xfunc_dpkg_installed_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -n -s Package
-    }
-
-    _comp_xfunc_dpkg_purgeable_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -o -FStatus 'ok config-files' -n -s Package
-    }
-else
-    _comp_xfunc_dpkg_installed_packages()
-    {
+_comp_xfunc_dpkg_installed_packages()
+{
+    local pkgs
+    pkgs=$(
+        grep-status -P -e "^${cur-}" -a \
```

Comment:
I don't remember if we had a rule against just trying to invoke commands that might not be present, e.g. for command not found handlers. If yes, I think we should document it somewhere.

Reply:
I don't remember if it is documented, but there are already many cases that try to call may-not-exist commands with `2>/dev/null`, so I guess it should be allowed.

As for the command not found handlers, we actually require the writers of `command_not_found_handler` to be responsible for it (by checking the existence of `COMP_LINE`, etc.) in README.md.